pkg/controllers/spannerautoscaler_controller.go

```diff
@@ -322,23 +336,36 @@ func (r *SpannerAutoscalerReconciler) needUpdateNodes(sa *spannerv1alpha1.Spanne
 	}
 }
 
+// For testing purpose only
 func calcDesiredNodes(currentCPU, currentNodes, targetCPU, minNodes, maxNodes, maxScaleDownNodes int32) int32 {
-	totalCPU := currentCPU * currentNodes
-	desiredNodes := totalCPU/targetCPU + 1 // roundup
+	return calcDesiredProcessingUnits(currentCPU, currentNodes*1000, targetCPU, minNodes*1000, maxNodes*1000, maxScaleDownNodes) / 1000
+}
+
+func roundUpProcessingUnits(processingUnits int32) int32 {
+	if processingUnits >= 1000 {
+		return ((processingUnits / 1000) + 1) * 1000
+	}
+	return ((processingUnits / 100) + 1) * 100
```

Comment:
This will always add 100 processing units to the input, even if the input is a multiple of 100. For example, `roundUpProcessingUnits(2600)` will return 2700, while it should ideally return just 2600.

Reply:
I agree for round-up function but the current behavior does always increment 1 node(1000 processing units) even if totalCPU is a multiple of targetCPU.
https://github.com/mercari/spanner-autoscaler/blob/ab62d5b4981e7fc2dc0917348a0160e12cc5f062/pkg/controllers/spannerautoscaler_controller.go#L326-L327

I think it is better to align with current behavior but renaming may be preferred.

Reply:
I will rename it to `nextPossibleProcessingUnits`

Reply:
Renamed in https://github.com/mercari/spanner-autoscaler/pull/34/commits/5aa8f468d3979a096bf9260ae804dc3c8156945f

Reply:
Thank you, it is clear now.